modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -0,0 +1,55 @@
+/** Rosetta Build Once. Integrate Your Blockchain Everywhere.
+  *
+  * The version of the OpenAPI document: 1.4.12 Contact: team@openapitools.org
+  *
+  * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech). https://openapi-generator.tech
+  */
+
+package org.tessellation.rosetta.domain
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.currency.Currency
+import derevo.circe.magnolia._
+import derevo.derive
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.boolean.Not
+import io.estatico.newtype.macros.newtype
+import io.circe.refined._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.{Decoder, Encoder}
+import org.tessellation.ext.refined.{decoderOf, encoderOf}
+import io.circe._
+
+object amount {
+  @derive(customizableDecoder, customizableEncoder)
+  case class Amount(
+    value: AmountValue,
+    currency: Currency
+  )
+
+  type AmountValueRefined = Long Refined Not[0L]
+
+  object AmountValue {
+    implicit val amountValueRefinedEncoder: Encoder[AmountValueRefined] =
+      encoderOf[Long, Not[0L]]
+
+    implicit val amountValueRefinedDecoder: Decoder[AmountValue] =
```

Comment:
I am working on getting the `Decoder` up and running. I tried `Validate.fromPredicate`, but I am not sure what the third argument should be. Pointers here are greatly appreciated.